### DISCUSSION TITLE
Better Google Analytics and other external tracking tools integration

### DISCUSSION THREAD

**Discussion by danicuki** [forem/forem#16587] [Q&A] [answered]
I would like to understand how can I use Google Analytics integration better with forem. I feel it a little bit limited. I'd like to:

 - track custom events on Google Analytics (button clicks)?
 - how to track when a user clicked a "Display Ads" that points to an external website?
 - Is there a way to use Google Tag Manager to forem?
 - Useful tools like Mixpanel, Hotjar, etc. I was wandering if there is a way to add custom javascript to my forem self-hosted instalation.

**Comment by JennieOcken** [comment, marked as answer]
Hi @danicuki thank you for the request.  I am working through cleaning up our discussions.

Right now, we don't have a lot of deep GA integration and I totally get the request.  Here is our plan:

We have a different discussion about some other [Analytics services](https://github.com/forem/forem/discussions/17200).

Plus, for allowing a Creator to add their own JavaScript or HTML to the header of the page, we actually have a https://github.com/forem/forem/issues/16333 for this right now. It is on our "high impact, small effort" list. Which means we slip in these tasks as our team has time or look for contributors to help us out with these features.

I'm going to remove this discussion as a duplicate.

**Comment by nickrider290** [comment]
Hi,
Nick here i have same question how can I use Google Analytics integration better with forem[.](https://www.dcfont.com/special-letters-generator/)I feel it a little bit limited.
